diff -ru lua-5.1.5/src/lvm.c lua-5.1.5-changes/src/lvm.c
--- lua-5.1.5/src/lvm.c	2011-08-17 22:43:11.000000000 +0200
+++ lua-5.1.5-changes/src/lvm.c	2025-05-26 15:49:48.011402537 +0200
@@ -205,8 +205,14 @@
   size_t ll = ls->tsv.len;
   const char *r = getstr(rs);
   size_t lr = rs->tsv.len;
+  //SPRING
+  const size_t n = std::min(ll, lr) + 1;
+  //SPRING
   for (;;) {
-    int temp = strcoll(l, r);
+    //SPRING
+    //int temp = strcoll(l, r);
+    int temp = strncmp(l, r, n);
+    //SPRING
     if (temp != 0) return temp;
     else {  /* strings are equal up to a `\0' */
       size_t len = strlen(l);  /* index of first `\0' in both strings */
