diff -ru lua-5.1.5/src/loslib.c lua-5.1.5-changes/src/loslib.c
--- lua-5.1.5/src/loslib.c	2008-01-18 17:38:18.000000000 +0100
+++ lua-5.1.5-changes/src/loslib.c	2025-05-26 16:18:14.567914352 +0200
@@ -70,9 +70,19 @@
   return 1;
 }
 
+//SPRING
+#ifndef BUILDING_AI
+#include "System/Misc/SpringTime.h"
+#endif
 
 static int os_clock (lua_State *L) {
-  lua_pushnumber(L, ((lua_Number)clock())/(lua_Number)CLOCKS_PER_SEC);
+  //SPRING
+#ifndef BUILDING_AI
+  //Lua's used clock() ran too fast on some linux systems, so rely on spring_time instead
+  lua_pushnumber(L, spring_time::getelapsedtime().toSecsf());
+#else
+  lua_pushnumber(L, (lua_Number)clock()/(lua_Number)CLOCKS_PER_SEC);
+#endif
   return 1;
 }
 
