diff -ru lua-5.1.5/src/luaconf.h lua-5.1.5-changes/src/luaconf.h
--- lua-5.1.5/src/luaconf.h	2008-02-11 17:25:08.000000000 +0100
+++ lua-5.1.5-changes/src/luaconf.h	2025-05-26 16:29:01.910908088 +0200
@@ -501,8 +501,10 @@
 ** ===================================================================
 */
 
-#define LUA_NUMBER_DOUBLE
-#define LUA_NUMBER	double
+//SPRING#define LUA_NUMBER_DOUBLE
+//SPRING#define LUA_NUMBER	double
+#undef LUA_NUMBER_DOUBLE
+#define LUA_NUMBER	float
 
 /*
 @@ LUAI_UACNUMBER is the result of an 'usual argument conversion'
@@ -518,7 +520,8 @@
 @@ LUAI_MAXNUMBER2STR is maximum size of previous conversion.
 @@ lua_str2number converts a string to a number.
 */
-#define LUA_NUMBER_SCAN		"%lf"
+//SPRING#define LUA_NUMBER_SCAN		"%lf"
+#define LUA_NUMBER_SCAN		"%f"
 #define LUA_NUMBER_FMT		"%.14g"
 #define lua_number2str(s,n)	sprintf((s), LUA_NUMBER_FMT, (n))
 #define LUAI_MAXNUMBER2STR	32 /* 16 digits, sign, point, and \0 */
